src/main/java/net/snowflake/ingest/streaming/internal/ParquetFlusher.java

```diff
@@ -39,23 +23,36 @@
 public class ParquetFlusher implements Flusher<ParquetChunkData> {
   private static final Logging logger = new Logging(ParquetFlusher.class);
   private final MessageType schema;
+  private final boolean enableParquetMemoryOptimization;
 
-  /** Construct parquet flusher from its schema. */
-  public ParquetFlusher(MessageType schema) {
+  /**
+   * Construct parquet flusher from its schema and set flag that indicates whether Parquet memory
+   * optimization is enabled.
+   */
+  public ParquetFlusher(MessageType schema, boolean enableParquetMemoryOptimization) {
     this.schema = schema;
+    this.enableParquetMemoryOptimization = enableParquetMemoryOptimization;
   }
 
   @Override
   public SerializationResult serialize(
-      List<ChannelData<ParquetChunkData>> channelsDataPerTable,
-      ByteArrayOutputStream chunkData,
-      String filePath)
+      List<ChannelData<ParquetChunkData>> channelsDataPerTable, String filePath)
+      throws IOException {
+    if (enableParquetMemoryOptimization) {
+      return serializeWithEnabledOptimization(channelsDataPerTable, filePath);
+    }
+    return serializeWithDisabledOptimization(channelsDataPerTable, filePath);
+  }
+
+  private SerializationResult serializeWithEnabledOptimization(
```

Comment:
```suggestion
  private SerializationResult serializeFromJavaObjects(
```
I think this would be more descriptive regarding what is going on :)

Reply:
I agree, renaming`serializeWithEnabledOptimization` -> `serializeFromParquetWriteBuffers `
`serializeWithDisabledOptimization ` -> `serializeFromJavaObjects`